lib/perl/Genome/Qc/Command/BuildMetrics.pm

```diff
@@ -118,4 +144,135 @@ sub _calculate_haploid_coverage {
     return 1;
 }
 
+
+sub output_metrics_as_tsv {
+    my $self = shift;
+    my $qc_metric_results = shift;
+    
+    my @headers = qw/
+                       sample_name
+                       build_id
+                       instrument_data_count
+                       instrument_data_ids
+                       bam_path
+                       pf_reads
+                       aligned_bases
+                       alignment_rate
+                       duplication_rate
+                       first_of_pair_mismatch_rate
+                       second_of_pair_mismatch_rate
+                       microarray_chipmix
+                       microarray_freemix
+                   /;
+
+    # The results themselves have a type, eg. wgs or exome
+    # Rather than relying on a user input profile, could use the result type instead
+    my @data;
+    if ($self->_tsv_output_profile eq 'wgs') {
+        #TODO: Add additional WGS specific metrics?
+        push @headers, qw/
+                             chipmix
+                             freemix
+                             haploid_coverage
+                         /;
+        for my $qc_metric_result (@$qc_metric_results) {
+            my $data = $self->_base_hash_ref_for_qc_metric_result($qc_metric_result);
+            $data->{'chipmix'} = $qc_metric_result->{ALL}->{CHIPMIX};
+            $data->{'freemix'} = $qc_metric_result->{ALL}->{FREEMIX};
+            $data->{'haploid_coverage'} = $qc_metric_result->{HAPLOID_COVERAGE};
+            $self->_add_microarray_vbid_result_metrics_to_hash_ref($qc_metric_result,$data);
+            push @data, $data;
+        }
+    } elsif ($self->_tsv_output_profile eq 'exome') {
+        push @headers, qw/
+                             mean_target_coverage
+                             pct_usable_bases_on_target
+                             pct_target_bases_20x
+                             pct_exc_off_target
+                             pct_exc_dupe
+                             pct_exc_overlap
+                             pct_exc_mapq
+                             pct_exc_baseq
+                         /;
+        for my $qc_metric_result(@$qc_metric_results) {
+            my $data = $self->_base_hash_ref_for_qc_metric_result($qc_metric_result);
+            $data->{'mean_target_coverage'} = $qc_metric_result->{MEAN_TARGET_COVERAGE};
+            $data->{'pct_usable_bases_on_target'} = $qc_metric_result->{PCT_USABLE_BASES_ON_TARGET};
+            $data->{'pct_target_bases_20x'} = $qc_metric_result->{PCT_TARGET_BASES_20X};
+            $data->{'pct_exc_off_target'} = $qc_metric_result->{PCT_EXC_OFF_TARGET};
+            $data->{'pct_exc_dupe'} = $qc_metric_result->{PCT_EXC_DUPE};
+            $data->{'pct_exc_overlap'} = $qc_metric_result->{PCT_EXC_OVERLAP};
+            $data->{'pct_exc_mapq'} = $qc_metric_result->{PCT_EXC_MAPQ};
+            $data->{'pct_exc_baseq'} = $qc_metric_result->{PCT_EXC_BASEQ};
```

Comment:
It's not quite that simple, but yes all the exome metric hash keys do happen to be the uppercase of the column header.  I made a change to reduce the redundancy.